dev/devicelab/lib/framework/adb.dart

```diff
@@ -392,9 +392,35 @@ class IosDeviceDiscovery implements DeviceDiscovery {
     _workingDevice = allDevices[math.Random().nextInt(allDevices.length)];
   }
 
+  // Returns the path to cached binaries relative to devicelab directory
+  String get _artifactDirPath {
+    return path.normalize(
+      path.join(
+        path.current,
+        '../../bin/cache/artifacts',
+      )
+    );
+  }
+
+  // Returns a colon-separated environment variable that contains the paths
+  // of linked libraries for idevice_id
+  Map<String, String> get _ideviceIdEnvironment {
+    final String libPath = const <String>[
+      'libimobiledevice',
+      'usbmuxd',
+      'libplist',
+      'openssl',
+      'ideviceinstaller',
+      'libtasn1',
+      'ios-deploy',
+    ].map((String packageName) => path.join(_artifactDirPath, packageName)).join(':');
+    return <String, String>{'DYLD_LIBRARY_PATH': libPath};
+  }
+
   @override
   Future<List<String>> discoverDevices() async {
-    final List<String> iosDeviceIDs = LineSplitter.split(await eval('idevice_id', <String>['-l']))
+    final String ideviceIdPath = path.join(_artifactDirPath, 'libimobiledevice', 'idevice_id');
```

Comment:
Use `artifacts.getArtifactPath` for this one as well.

Reply:
I don't think I can, as this is the devicelab project, and it has no access to the flutter_tools code. Is that right @jonahwilliams?

Reply:
This package doesn't currently depend on the flutter tool. We could add a dependency here, but that would still require a `src` import. I'm okay with using the hardcoded path since this is for our own internal testing framework

Reply:
Oh I didn't realize this is in the test code. In that case I'm fine with that.